activerecord/lib/active_record/connection_adapters/sqlite3/quoting.rb

```diff
@@ -13,7 +13,7 @@ def quote_table_name_for_assignment(table, attr)
         end
 
         def quote_column_name(name)
-          @quoted_column_names[name] ||= %Q("#{super.gsub('"', '""')}")
+          @quoted_column_names[name] ||= %Q("#{super.gsub('"', '""').gsub('.', '"."')}")
```

Comment:
For consistency, how about moving the new gsub to `quote_table_name`?

https://github.com/rails/rails/blob/647665110f49935b30e29ac09ecfca3701e76041/activerecord/lib/active_record/relation/query_methods.rb#L1057-L1058
https://github.com/rails/rails/blob/647665110f49935b30e29ac09ecfca3701e76041/activerecord/lib/active_record/connection_adapters/mysql/quoting.rb#L11-L13

Reply:
Done! Good catch